ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/workspace/WorkspaceEventsHandler.java

```diff
@@ -220,17 +226,27 @@ public void apply(WorkspaceDto workspace) throws OperationException {
      */
     private void restoreDevMachineLogs(final Machine devMachine) {
         if (devMachine != null) {
-            machineServiceClient.getProcesses(devMachine.getWorkspaceId(), devMachine.getId())
-                                .then(new Operation<List<MachineProcessDto>>() {
-                                    @Override
-                                    public void apply(List<MachineProcessDto> arg) throws OperationException {
-                                        for (MachineProcessDto process : arg) {
-                                            if ("CheWsAgent".equals(process.getName())) { //get start ws agent process
-                                                getLogsOfWsAgent(process, devMachine);
-                                            }
-                                        }
-                                    }
-                                });
+            final String devMachineId = devMachine.getId();
+            execAgentCommandManager.getProcesses(devMachineId, false).then(new Operation<List<GetProcessesResponseDto>>() {
+                @Override
+                public void apply(List<GetProcessesResponseDto> processes) throws OperationException {
+                    for (GetProcessesResponseDto process : processes) {
+                        String candidateName = process.getName();
+                        String name = "CheWsAgent";
```

Comment:
`CheWsAgent` might be extracted to the shared constant. To avoid hardcoding it here.